dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/HttpsProxyDefaultRoutePlanner.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.hc5;
+
+import com.palantir.conjure.java.client.config.HttpsProxies;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.ProxySelector;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import org.apache.hc.client5.http.impl.routing.DefaultRoutePlanner;
+import org.apache.hc.client5.http.impl.routing.SystemDefaultRoutePlanner;
+import org.apache.hc.core5.http.HttpException;
+import org.apache.hc.core5.http.HttpHost;
+import org.apache.hc.core5.http.protocol.HttpContext;
+
+/**
+ * Identical to {@link SystemDefaultRoutePlanner} but adds support for connecting to an HTTPS proxy.
```

Comment:
Let's link the original source [here](https://github.com/apache/httpcomponents-client/blob/rel/v5.3.1/httpclient5/src/main/java/org/apache/hc/client5/http/impl/routing/SystemDefaultRoutePlanner.java) and describe that the original is licensed under apache 2.0.

Reply:
A bit confused, it is linked, did you want it linked differently? Or you want a link to http://www.apache.org/licenses/LICENSE-2.0

Added a line below "Original version is licensed under Apache License, Version 2.0."

Reply:
I'd like to have a link to the original, at the version used at the point in time we modified from. This way if there are future changes, we can tell whether or not they may be impactful. The javadoc link will point toward whatever version we happen to depend on at any point in time

Reply:
Makes sense, updated the docs